MoreLinq.Test/FullGroupJoinTest.cs

```diff
@@ -102,29 +108,39 @@ public void FullGroupPreservesOrder()
                 (1, 4),
                 (3, 3),
             };
-            var listB = new[] {
+            var listB = new[]
+            {
                 (4, 1),
                 (3, 1),
                 (2, 1),
                 (0, 1),
                 (3, 0),
             };
 
-            var result = listA.FullGroupJoin(listB, x => x.Item1, x => x.Item1, (key, first, second) => new { key, first, second }).ToList();
+            var result = FullGroupJoin(overloadCase, listA, listB, x => x.Item1).ToList();
 
             // Order of keys is preserved
-            result.Select(x => x.key).AssertSequenceEqual(3, 1, 2, 4, 0);
+            result.Select(x => x.Key).AssertSequenceEqual(3, 1, 2, 4, 0);
 
             // Order of joined elements is preserved
-            foreach (var res in result) {
-                res.first.AssertSequenceEqual(listA.Where(t => t.Item1 == res.key).ToArray());
-                res.second.AssertSequenceEqual(listB.Where(t => t.Item1 == res.key).ToArray());
+            foreach (var res in result)
```

Comment:
Non-essential but perhaps this can be simplified to?

```c#
            foreach (var (key, fst, snd) in result)
            {
                fst.AssertSequenceEqual(listA.Where(t => t.Item1 == key).ToArray());
                snd.AssertSequenceEqual(listB.Where(t => t.Item1 == key).ToArray());
            }
```